portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -95,14 +109,22 @@ public void doDelete(HttpServletRequest request, HttpServletResponse response) t
     query.addSort(sortOrder.property, sortOrder.sortDirection);
 
     // Create comment list
-    PreparedQuery results = datastore.prepare(query);
+    PreparedQuery preparedQuery = datastore.prepare(query);
+
+    QueryResultList<Entity> results = preparedQuery.asQueryResultList(fetchOptions);
 
     ArrayList<Comment> comments = new ArrayList<>();
-    for (Entity entity: results.asIterable(fetchOptions)) {
+    for (Entity entity: results) {
       Comment comment = transformEntityToComment(entity);
       comments.add(comment);
     }
-    return comments;
+
+    if (comments.size() == 0) {
+      return new CommentsResult();
```

Comment:
Either take in isEndOfComments explicitly as an argument in your constructor or use a factory method for clarity that this is going to flip isEndOfComments to true?